src/Orleans/Threading/CancellationTokenHolderGrain.cs

```diff
@@ -0,0 +1,50 @@
+﻿using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Orleans.Placement;
+
+namespace Orleans.Threading
+{
+    [PreferLocalPlacement]
+    internal class CancellationTokenHolderGrain : Grain, ICancellationTokenHolderGrain
+    {
+        private CancellationTokenSource cancellationTokenSource;
+        private bool collectionScheduled;
+
+        public override Task OnActivateAsync()
+        {
+            DelayDeactivation(TimeSpan.FromDays(5));
+            cancellationTokenSource = new CancellationTokenSource();
+            return TaskDone.Done;
+        }
+
+        public Task<CancellationToken> GetCancellationToken()
+        {
+            return Task.FromResult(cancellationTokenSource.Token);
+        }
+
+        public Task Cancel()
+        {
+            cancellationTokenSource.Cancel();
+            DelayDeactivation(TimeSpan.FromSeconds(30));
```

Comment:
Will add additional number to the https://github.com/dotnet/orleans/pull/1569/files#diff-7ddbc03b5e6c00821897b355928dda00R14